ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -173,6 +173,8 @@ void resume() {
     hardDeleteLock.lock();
     try {
       if (paused.compareAndSet(true, false)) {
+        startToken =
```

Comment:
I think this is unrelated to this change and the check you suggest should be added as a test assert in the hard delete pause-resume test in `IndexTest`. When you find the time, could you add the assertion?